docs/onboarding-guide.md

```diff
@@ -0,0 +1,171 @@
+
+* **[CommandHandlerInput](https://github.com/aws/aws-dotnet-deploy/tree/main/src/AWS.Deploy.CLI/Commands/CommandHandlerInput)** - This contains the POCO objects that hold the values for all CLI switches of the associated command. These objects are provided as input while creating the command handlers. See [here](https://github.com/aws/aws-dotnet-deploy/blob/66fb1e34bbea10bbd995596e1eec801b79ff9102/src/AWS.Deploy.CLI/Commands/CommandFactory.cs#L185) for an example.
+* **[TypeHintCommandFactory](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.CLI/Commands/TypeHints/TypeHintCommandFactory.cs)** - Some of the option settings require special logic on how to prompt the user for a value and how to parse the user input. This class stores the mapping between the option settings and their corresponding typehints. **TypeHints** can be found [here](https://github.com/aws/aws-dotnet-deploy/tree/main/src/AWS.Deploy.CLI/Commands/TypeHints). **TypeHintResponses** can be found [here](https://github.com/aws/aws-dotnet-deploy/tree/main/src/AWS.Deploy.CLI/TypeHintResponses)
+* [**ConsoleUtilities**](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.CLI/ConsoleUtilities.cs) - Contains utility methods to drive the CLI UX
+* [**CommandLineWrapper**](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.CLI/Utilities/CommandLineWrapper.cs) - Contains utility methods to invoke command line processes like `docker build` or `cdk deploy`
+* [**Exceptions**](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.CLI/Exceptions.cs) - Declares all exceptions that are thrown by the AWS.Deploy.CLI component
+* [**AWS.Deploy.CLI.UnitTests**](https://github.com/aws/aws-dotnet-deploy/tree/main/test/AWS.Deploy.CLI.UnitTests) - Contains unit tests for this component
+
+
+
+### [AWS.Deploy.Recipes](https://github.com/aws/aws-dotnet-deploy/tree/main/src/AWS.Deploy.Recipes)
+
+This component contains the recipes that serve as the backbone of each deployment recommendation. It also contains the CDK templates that is used to deploy the customer’s .NET application as a CloudFormation stack
+
+#### Key Components
+
+* **[RecipeDefinitions](https://github.com/aws/aws-dotnet-deploy/tree/main/src/AWS.Deploy.Recipes/RecipeDefinitions)** - It contains the deployment recipes that target different AWS services. Each recipe file contains a list of option settings which is a collection of AWS resources that users can customize as per their needs. 
```

Comment:
You might want to also link to the reference docs on the recipe schema https://aws.github.io/aws-dotnet-deploy/docs/deployment-projects/recipe-file/